addons/binding/org.openhab.binding.wink/src/main/java/org/openhab/binding/wink/handler/ThermostatHandler.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.wink.handler;
+
+import static org.openhab.binding.wink.WinkBindingConstants.*;
+
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.wink.client.IWinkDevice;
+import org.openhab.binding.wink.client.WinkSupportedDevice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joel Shuman - Initial contribution
+ */
+public class ThermostatHandler extends WinkBaseThingHandler {
+	private final Logger logger = LoggerFactory.getLogger(ThermostatHandler.class);
+
+	public ThermostatHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void handleWinkCommand(ChannelUID channelUID, Command command) {
+		if (command instanceof RefreshType) {
+			logger.debug("Refreshing state");
+			updateDeviceState(getDevice());
+		}
+		if (channelUID.getId().equals(CHANNEL_THERMOSTAT_CURRENTSETPOINT)) {
+			if (command instanceof Number) {
+				logger.debug("Setting desired temperature {}", command);
+				float temperature = ((Number) command).floatValue();
+				setDesiredTemperature(temperature);
+
+				// Try running the update after a few seconds.
+				delayedUpdateState();
+			}
+		} else if (channelUID.getId().equals(CHANNEL_THERMOSTAT_CURRENTMODE)) {
+			if (command instanceof StringType) {
+				logger.debug("Setting desired mode {}", command);
+				String mode = ((StringType) command).toString();
+				setDesiredMode(mode);
+
+				// Try running the update after a few seconds.
+				delayedUpdateState();
+			}
+		}
+	}
+	
+	@Override
+	protected boolean connectionStatus(IWinkDevice device){
+		Map<String, String> jsonData = getDevice().getCurrentStateComplexJson();
+		return jsonData.get("connection").equals("true");
+	}
+
+	private void setDesiredTemperature(float temperature) {

```

Comment:
Move all the thermostat specific logic from the bridge handler to these methods.